server/modifier_test.go

```diff
@@ -65,3 +65,55 @@ func TestIntegration_ResponseHeaders(t *testing.T) {
 		})
 	}
 }
+
+func TestIntegration_SetResponseStatus(t *testing.T) {
+	const confFile = "testdata/integration/modifier/02_couper.hcl"
+
+	client := newClient()
+	helper := test.New(t)
+
+	type testCase struct {
+		path       string
+		expMessage string
+		expStatus  int
+	}
+
+	for _, tc := range []testCase{
+		{
+			path:       "/204",
+			expMessage: "set_response_status sets the HTTP status code to 204 - removing the response body if any",
+			expStatus:  204,
+		},
+		{
+			path:       "/201",
+			expMessage: "",
+			expStatus:  201,
+		},
+		{
+			path:       "/600",
+			expMessage: "configuration error: set_response_status sets an invalid HTTP status code: 600; set the status code to 500",
+			expStatus:  500,
+		},
+	} {
+		t.Run(tc.path, func(subT *testing.T) {
+
+			shutdown, hook := newCouper(confFile, helper)
+			defer shutdown()
+
+			req, err := http.NewRequest(http.MethodGet, "http://example.com:8080"+tc.path, nil)
+			helper.Must(err)
```

Comment:
Using the `helper` with parent `t` will mark the parent test as failed. Use a new one with `subT`.

Reply:
Done